src/interfaces/simulation.ts

```diff
@@ -224,11 +224,16 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     vaultContract: VaultContract,
     options: SimulationOptions
   ): Promise<TransactionOutcome> {
-    const encodedInputData = vaultContract.encodeDeposit(amount);
+    const encodedInputData =
+      this.yearn.services.partner && this.yearn.services.partner.isAllowed(toVault)
```

Comment:
Logic (https://github.com/yearn/yearn-sdk/pull/224/files#r822297451) is repeated in 3 places

Reply:
solved on 64bdcb84424cdca052f18fbae628b30e089b0076